packages/block-library/src/navigation/edit/create-new-menu-button.js

```diff
@@ -0,0 +1,27 @@
+/**
+ * WordPress dependencies
+ */
+import { Button } from '@wordpress/components';
+import { useState } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+
+const CreateNewMenuButton = ( { onCreateNew } ) => {
+	const [ isCreatingMenu, setIsCreatingMenu ] = useState( false );
+
+	return (
+		<Button
+			className="wp-block-navigation__navigation-selector-button--createnew"
+			isBusy={ isCreatingMenu }
+			disabled={ isCreatingMenu }
+			__experimentalIsFocusable
+			onClick={ () => {
+				onCreateNew();
+				setIsCreatingMenu( true );
```

Comment:
It is unusual that we need an internal var for this. We can't pass the state down into the component?

Reply:
I was trying to avoid passing props around. What would be the benefit of sharing state here? I believe it's only used to control the UI.

Reply:
We're using internal state that isn't connected to the actual result of the action we're invoking.

That _could_ leave our button in a permanent "loading" state. We're relying on the component being unmounted in order to reset the internal state back to not loading. If this button ends up in a different context to that which you are expecting here it may introduce bugs.

An alternative would be to pass the resolution state down into the component and use that to conditionalise the internal UI.

Currently in this example the function that ends up being called is

```js
const createUntitledEmptyNavigationMenu = () => {
	createNavigationMenu( '' );
};
```

Here the error state isn't being handled inline at all but rather it's deferred to the `useEffect`s in the main component. We should have a `.catch` (that's for another PR).

The point is that `createNavigationMenu` could easily error and the `CreateNewMenuButton` would know nothing about it and could potentially end up loading forever.

Either the component is controlled or it is uncontrolled. I feel like it's currently a bit of both.